### ISSUE TITLE
TypeVar replacement type becomes `Any` with new style union

### ISSUE THREAD

**Issue by layday** [microsoft/pyright#1180]
**Describe the bug**
When using a vertical bar to express a union between a type variable and another type, the resolved type is `Any | other_type` for any value of `T` in `T | other_type`.

**To Reproduce**
Type check the snipper below.

**Expected behavior**
That the type of `foo('')` would be `str | None`, etc.

**Screenshots or Code**

```python
# pyright: strict

from __future__ import annotations

from typing import Generic, TypeVar, Union


T = TypeVar('T')

def foo(value: T) -> T | None:
    ...

def bar(value: T) -> Union[T, None]:
    ...

class Baz(Generic[T]):
    qux: T | None


reveal_type(foo(''))  # Type of "foo('')" is "Any | None"
reveal_type(bar(''))  # Type of "bar('')" is "Literal[''] | None"
reveal_type(Baz[str].qux)  # Type of "Baz[str].qux" is "Any | None"

```

**VS Code extension or command-line**
Either, 1.1.86

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is now addressed in pyright 1.1.87, which I just published. It will also be included in the next release of pylance.
